### PR TITLE
Update makefile colorize tests according to the latest grammar

### PR DESCRIPTION
**Pull request by fadeevab**
1. Handling the '!=' variable assignment operator.
2. Handling the "#" (hashtag) in the quotes inside the recipe.
3. Restore the recipe handling.

Issue: #60030, #61475

Recipe as it was:
makefile-grammar-shell-color-prev

Recipe is looking good now:
makefile-grammar-shell-color

### CODE REVIEW

**Comment by fadeevab** [comment]
@aeschli 
The pull request is about to fix 2 requested issues + 1 minor issue I found + improvement of recipes.

Test cases will pass after updating the makefile grammar from the upstream (https://github.com/fadeevab/make.tmbundle/). Thank you!

**Comment by alexr00** [comment]
@fadeevab, thanks for adding the test case! I'll pull in your grammar update and merge this in next week for the November 2018 milestone.

File: extensions/make/test/colorize-results/makefile.json [hunk 1]
```diff
@@ -342,7 +342,7 @@
 	},
 	{
 		"c": "\tg++ main.o factorial.o hello.o -o hello",
-		"t": "source.makefile",
+		"t": "source.makefile meta.scope.recipe.makefile",
```
Comment (zoe704) [inline extensions/make/test/colorize-results/makefile.json]: #

**Review by alexr00 — APPROVED** [review]
Thank you for the PR!

**Comment by fadeevab** [comment]
@alexr00 You're welcome!

**Comment by fadeevab** [comment]
Just for information

How it was:
makefile-grammar-shell-color-prev

Recipe is looking good now:
makefile-grammar-shell-color

And a few other fixes.

**Pull request merged**